clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3288,7 +3351,7 @@ namespace {
         } else {
           BoundsExpr *NormalizedDeclaredBounds = ExpandToRange(D, DeclaredBounds);
```

Comment:
I believe the observed bounds need to be set in this case.   Please open an issue to track this, as a separate follow-up.  Addressing this case I believe will handled the TODO that is just above (at line 3273).